### DISCUSSION TITLE
🎚 Multiple audio output control

### DISCUSSION THREAD

**Discussion by Bryceed** [ppy/osu#13233] [Ideas]
Many people like to play with the hitsounds louder than the music, others think it "too loud" and enjoy more listening to the beatmap music than the hitting sounds.

One of the scenarios that this happens most is of those who record or stream osu! where, for those who play (generally) it's better scenario to "feel the hitsound" while, for those who watch, it would be more pleasant to have this more balanced sound, among other cases.

So, **osu!lazer could offer an option of multi outputs to in-game sounds control**, this would help mainly for those who make lives to offer a better volume control to the audience without compromising the gameplay of the streamer.

**Comment by peppy** [comment]
Hugely disagree with this. Seems disingenuous and weird that you'd stream different audio out to viewers than what you're using the play the game. Would be like streaming the gameplay with a different skin to what you're using.

Also this would be a huge amount of extra effort for very little gain. Don't foresee this happening unless someone as competent as those on the core team takes this on themselves.

**Reply by Bryceed** [reply]
The question isn't to reproduce different audios (as a "equivalence" to using different skins to different video outputs), but to have the ability to better adjust the volume of the game, which would be good for streamers to be able to make a better adjustment to the game sound in their lives. A video equivalent would be like being able to adjust the dim and the blurry background (like 100% black for the streamer and a 70% + storyboard/video: ON for streaming output, which would be nice too but this is not the case).

But I understand that such an implementation could impact even in volume adjustments in the UI, I just think it would be interesting if the game offered this level of adjustment for those who thought it better this way (and not just being able to increase or decrease the volume of everything in the lives/records). 

_It may not really be used only for "normal players", but for streamers, having that level of sound control could make a big difference, especially when playing maps that hitsounds are extremely loud (which can be good for the accuracy of the streamer, but it can be very uncomfortable for those who watch)._

**Reply by peppy** [reply]
We already plan on offering a separate volume for gameplay and UI samples. Also we have an ongoing effort to normalise volume across beatmaps, which should help with the scenario you mention.

**Comment by smoogipoo** [comment]
Outputting to different outputs would be helpful for livestreamers playings in tournaments too. That way you could talk with your teammates but still stream your perspective without worrying the other team's listening in on tactics. Routing solutions like VAC/VB introduce latency.

It's quite a fringe purpose, and I'd say this is very low priority though.

**Reply by peppy** [reply]
Isn't that something you'd configure in discord not osu!..? This is about changing the audio and sample outputs individually, which would not work around the issue you described (and still require virtual devices).

**Reply by smoogipoo** [reply]
How would you change that in discord? OBS would have to attach to the output device if you wanted lowest latency (osu! -> output).

What you'd want is the following setup:
```
osu!
  -> VAC
    -> Output
    -> OBS
```
Where you can then point Discord to the output and hide it from OBS. But as I said this can't be achieved without incurring VAC latency.

> This is about changing the audio and sample outputs individually

Yeah, I was just thinking on a broader scope with such a generic title.
